File: leet-code/medium/0102 binary-tree-level-order-traversal.go
package leet

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func levelOrder(root *TreeNode) [][]int {
	if root == nil {
		return nil
	}

	level := [][]int{{root.Val}}

	l := levelOrder(root.Left)
	r := levelOrder(root.Right)
	ls := len(l)
	lr := len(r)

	for i := 0; i < ls || i < lr; i++ {
		childs := []int{}
		if i < ls {
			childs = l[i]
		}

		if i < lr {
			childs = append(childs, r[i]...)
		}

		level = append(level, childs)
	}

	return level
}

File: leet-code/python/0102_binary_tree_level_order_traversal.py
# Definition for a binary tree node.
class TreeNode:
  def __init__(self, val=0, left=None, right=None):
    self.val = val
    self.left = left
    self.right = right
    
class Solution:
  def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
    levels = []
    def getLevelNodes(node, level):
      if not node:
        return
      
      if len(levels) < level:
        levels.append([])
      
      levels[level - 1].append(node.val)
      getLevelNodes(node.left, level + 1)
      getLevelNodes(node.right, level + 1)
    
    getLevelNodes(root, 1)
    return levels
        